File: src/main/java/com/anselumjuju/stores/TokenStore.java
package com.anselumjuju.stores;

public class TokenStore {
    private static String accessToken;
    private static String expiryTime;
    private static String workspaceId;

    public static String getAccessToken() {
        return accessToken;
    }

    public static void setAccessToken(String accessToken) {
        TokenStore.accessToken = accessToken;
    }

    public static String getExpiryTime() {
        return expiryTime;
    }

    public static void setExpiryTime(String expiryTime) {
        TokenStore.expiryTime = expiryTime;
    }

    public static String getWorkspaceId() {
        return workspaceId;
    }

    public static void setWorkspaceId(String workspaceId) {
        TokenStore.workspaceId = workspaceId;
    }

    public static boolean isTokenExpired(String expiryTime) {
        if (expiryTime == null || expiryTime.trim().isEmpty())
            return true;

        long expiryTimeLong = (long) Double.parseDouble(expiryTime);
        return System.currentTimeMillis() >= expiryTimeLong;
    }

}

File: backend/src/stores/tokenStore.js
let accessToken = null;
let expiryTime = null;

export function getAccessToken() {
  return accessToken;
}

export function setAccessToken(value) {
  accessToken = value;
}

export function getExpiryTime() {
  return expiryTime;
}

export function setExpiryTime(value) {
  expiryTime = value;
}

export function isTokenExpired(value) {
  if (value == null || String(value).trim().length === 0) return true;
  return Date.now() >= Number(value);
}
